### DISCUSSION TITLE
How to wrap Content in ContentTemplate

### DISCUSSION THREAD

**Discussion by SmRiley** [AvaloniaUI/Avalonia#13408] [Q&A] [answered]
In WPF, I can achieve this in the following way, but in Avalonia, I cannot successfully get the Content whether I define ContentComplate in the style or directly in the control.
```csharp
<Style TargetType="Button" x:Key="BtnIcon">
      <Setter Property="ContentTemplate">
	      <Setter.Value>
		      <DataTemplate>
			      <TextBlock Background="Red"
							       Text="{Binding}"></TextBlock>
		      </DataTemplate>
	      </Setter.Value>
      </Setter>
</Style>
```

**Comment by timunie** [comment]
Can you show your xaml of Avalonia please? It should work the same way, but proably DataType should be set for compiled bindings

**Reply by SmRiley** [reply]
this is my code:
![image](https://github.com/AvaloniaUI/Avalonia/assets/45205313/e8253698-bd4d-4fd8-a713-c9e607580158)

This is the running effect:
![image](https://github.com/AvaloniaUI/Avalonia/assets/45205313/d8d087b6-c29b-47db-92cc-47413214d5ce)

This is project file:
[AvaloniaApplication1.zip](https://github.com/AvaloniaUI/Avalonia/files/13191555/AvaloniaApplication1.zip)

**Reply by rabbitism** [reply, marked as answer]
DataType should be type of DataContext, but not type of control.

**Reply by SmRiley** [reply]
u are right , thank you , I guess it's because WPF can display the datatype correctly even if I fill in the wrong datatype, which makes me lose my way. can you explain this?

**Reply by SmRiley** [reply]
@rabbitism  Sorry, I think you have done a good job in helping me solve this problem. I should not ask for further explanation. I will look it up myself. Thank you again.

**Reply by timunie** [reply]
@SmRiley please lookup Avalonia.Samples for DataTemplate samples. 

https://github.com/AvaloniaUI/Avalonia.Samples

**Reply by SmRiley** [reply]
Thank you, I will do more detailed research and will not ask questions easily.

**Reply by rabbitism** [reply]
The idea is quite easy thou, data template deacribes what the data looks like as part of control, so datatype should be the data. 
Control template describes what a control should look like,  so target type should be control type.
